### PR TITLE
[RO-431] User cache warmup improvements

### PR DESCRIPTION
**Pull request by Skyld**
### Added

- Added more resiliency to user cache warmup process by introducing event dispatch retry mechanism in `UserCacheWarmerService` class.
- Added `USER_CACHE_WARMUP_MESSAGE_PAYLOAD_BATCH_SIZE` environment variable to be able to control the number of users to include per cache warmup message payload.
- Added `USER_CACHE_WARMUP_RETRY_WAIT_INTERVAL` environment variable to be able to control the waiting time interval between retry attempts during user cache warmup.

### Fixed

- Fixed scenario when user cache warmup is taking too much time to run in case of large volume of users due to poor DB query performance by removing unnecessary `distinct` call in `findAllUsernamesPaged` method of `UserRepository` class.

### CODE REVIEW

File: src/Service/Cache/UserCacheWarmerService.php [hunk 1]
```diff
@@ -30,20 +32,35 @@
...
     ) {
         $this->messageBus = $messageBus;
         $this->messengerLogger = $messengerLogger;
         $this->cacheWarmupLogger = $cacheWarmupLogger;
+
+        if ($cacheWarmupPayloadBatchSize < 1) {
+            throw new InvalidArgumentException('Message payload size must be greater or equal to 1.');
+        }
+
+        if ($userCacheWarmupRetryWaitInterval < 1) {
```
Comment (uncleempty) [inline src/Service/Cache/UserCacheWarmerService.php:58]: Though it's not a problem at all, but having minimum at 1 microsecond doesn't make sense :upside_down_face:

**Review by uncleempty — DISMISSED** [review]
:+1: 
Good job Dani!

File: src/Service/Cache/UserCacheWarmerService.php (hunk 1 above)
Comment (Skyld) [reply src/Service/Cache/UserCacheWarmerService.php]: Fair point, fixed :)

**Review by uncleempty — APPROVED** [review]
Thank you, @Skyld !
- [X] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Changelog is updated according to changes (if applicable)
- [X] Documentation is updated according to changes (if applicable)

Good job, just one moment:
> - [ ] New code is not subject to concurrency issues (if applicable)

This one is not ticked for only one reason - it's impossible really check on local to see if there any concurrency issues, it can be checked only with real numbers. BUT, since this changes are coming from what was discussed previously and already implemented as a workaround in prod - shouldn't be.

**Review by luisfn — APPROVED** [review]
Good job Dani

**Review Checklist**
- [X] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Changelog is updated according to changes (if applicable)
- [X] Documentation is updated according to changes (if applicable)

**Pull request merged**